### PR TITLE
Flink Operator on EKS 

### LINKED ISSUE #21: [Feature] Apache Flink on Amazon EKS
Build a Terraform template to build and deploy [Apache Flink](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/resource-providers/standalone/kubernetes/) on EKS

- Terraform Template for deployment
- If you need new Kubernetes add-ons then use [EKS Blueprints](https://github.com/aws-ia/terraform-aws-eks-blueprints) to add a new add-on
- README with deployment steps
- Add a new [Website Doc](https://github.com/awslabs/data-on-eks/tree/main/website/docs/streaming-platforms-eks) with design diagram, deployment (refer to README here for deployment steps) and how to run sample tests etc.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by vara-bonthu**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

- Flink Operator deployment with VPC, EKS, Node groups and Karpenter
- Flink job examples to use Managed Node groups and Node group less Karpenter
- Flink examples to use SSD storage
- Flink Website doc

Closes #21 

### Motivation

Closes #21 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [x] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: streaming/flink/flink-team.tf [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  description = "IAM role policy for flink Job execution"
+  name        = "${local.name}-flink-irsa"
+  policy      = data.aws_iam_policy_document.flink_operator.json
+}
+
+#---------------------------------------------------------------
+# IRSA for flink pods for "flink-team-a"
+#---------------------------------------------------------------
+module "flink_irsa" {
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/irsa?ref=v4.15.0"
```
Comment (askulkarni2) [inline streaming/flink/flink-team.tf:18]: Can we use `github.com/aws-ia/terraform-aws-eks-blueprints-addons//modules/irsa`  instead?

File: streaming/flink/helm-values/metrics-server-values.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# HA config for metrics-server
+image:
+  repository: k8s.gcr.io/metrics-server/metrics-server
```
Comment (askulkarni2) [inline streaming/flink/helm-values/metrics-server-values.yaml:3]: Use `registry.k8s.io` instead. See https://aws.amazon.com/blogs/containers/changes-to-the-kubernetes-container-image-registry/

File: streaming/flink/vpc.tf [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#---------------------------------------------------------------
+# Supporting Network Resources
+#---------------------------------------------------------------
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 3.0"
```
Comment (askulkarni2) [inline streaming/flink/vpc.tf:6]: `4.0` module is available and consolidates some of the code. See https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1546

**Review by askulkarni2 — APPROVED** [review]
Some optional minor comments, but otherwise lgtm!

File: streaming/flink/helm-values/metrics-server-values.yaml (hunk 2 above)
Comment (vara-bonthu) [reply streaming/flink/helm-values/metrics-server-values.yaml]: Thanks for this. Did a find and replace on this repo

File: streaming/flink/flink-team.tf (hunk 1 above)
Comment (vara-bonthu) [reply streaming/flink/flink-team.tf]: updated

File: streaming/flink/vpc.tf (hunk 3 above)
Comment (vara-bonthu) [reply streaming/flink/vpc.tf]: updated

**Pull request merged**